### DISCUSSION TITLE
How to provide multi label dataset for object detection?

### DISCUSSION THREAD

**Discussion by thhart** [deepjavalibrary/djl#176] [Q&A] [answered]
What is the expected structure of AccessDataset for training object detection models. The PikachuDetection contains a nice example however it is a one class example only. Below is the label extraction example constructing a Record structure. How must this structure constructed if there are multiple classes to be trained? Where is the class number to be specified?

```
    protected Record get(NDManager manager, long index) throws IOException {
        int idx = Math.toIntExact(index);
        NDList d =
                new NDList(
                        ImageFactory.getInstance()
                                .fromFile(imagePaths.get(idx))
                                .toNDArray(manager, flag));
        NDArray label = manager.create(labels.get(idx));
        NDList l = new NDList(label.reshape(new Shape(1).addAll(label.getShape())));
        return new Record(d, l);
    }

```

**Comment by stu1130** [comment, marked as answer]
The label type is `NDList`, which represents a list of NDArray. You can add several labels to construct a label NDList.
